File: latex/code/bi0_elf.m
% MATLAB/Octave code for Example 6

% Define the three models
e =  5;
s = 15;

h0 = @(x) (x<e & x>-e) / (2*e);
hn = @(x) normpdf(x,-e,s)*2.*(x<-e);
hp = @(x) normpdf(x,e,s)*2.*(x> e);

% Define the data and likelihood
d   =  -2;
n   = 100;
sem = sqrt((s^2 + s^2) / (2*n));

likelihood = @(x) normpdf(d,x,sem);

% Define the integrands and integrate
fn = @(x)likelihood(x).*hn(x);
mn = quadgk(fn,-inf,-e,'waypoints',[-e,e]);

f0 = @(x)likelihood(x).*h0(x);
m0 = quadgk(f0,-e,e,'waypoints',[-e,e]);

fp = @(x)likelihood(x).*hp(x);
mp = quadgk(fp,e,inf,'waypoints',[-e,e]);

ev = [mn,m0,mp];

% Apply Bayes' rule
eq19 = @(p,m) p.*m ./ sum(p.*m);

marchbanks = [.25,.50,.25];
granger    = [.15,.70,.15];
runcorn    = [.45,.10,.45];

eq19(marchbanks,ev)
% ans =    0.0061    0.9939    0.0000
eq19(granger,ev)
% ans =    0.0026    0.9974    0.0000
eq19(runcorn,ev)
% ans =    0.0122    0.9878    0.0000

File: latex/code/bi0_elf.R
# R code for Example 6

# Define the three models
e <-  5
s <- 15

h0 <- function(x) (x<e & x>-e) / (2*e)
hn <- function(x) dnorm(x,-e,s)*2*(x<-e)
hp <- function(x) dnorm(x, e,s)*2*(x> e)

# Define the data and likelihood
d   <-  -2
n   <- 100
sem <- sqrt((s^2 + s^2) / (2*n))

like <- function(x) dnorm(d,x,sem)

# Define the integrands and integrate
fn <- function(x) like(x)*hn(x)
mn <- integrate(fn,-Inf,-e)$value

f0 <- function(x) like(x)*h0(x)
m0 <- integrate(f0,-e,e)$value

fp <- function(x) like(x)*hp(x)
mp <- integrate(fp,e,Inf)$value

ev <- c(mn,m0,mp)

# Apply Bayes' rule
eq19 <- function(p,m) p*m / sum(p*m)

marchbanks <- c(.25,.50,.25)
granger    <- c(.15,.70,.15)
runcorn    <- c(.10,.10,.80)

eq19(marchbanks,ev)
# [1] 6.1457e-03 9.9385e-01 4.1385e-07
eq19(granger,ev)
# [1] 2.6432e-03 9.9736e-01 1.7799e-07
eq19(runcorn,ev)
# [1] 1.2216e-02 9.8778e-01 6.5810e-06
